pkg/provider/kubernetes/crd/kubernetes.go

```diff
@@ -586,20 +587,94 @@ func createDigestAuthMiddleware(client Client, namespace string, digestAuth *v1a
+func getBasicAuthCredentials(k8sClient Client, authSecret, namespace string) ([]string, error) {
 	if authSecret == "" {
 		return nil, fmt.Errorf("auth secret must be set")
 	}
 
-	auth, err := loadAuthCredentials(namespace, authSecret, k8sClient)
+	auth, err := loadBasicAuthCredentials(namespace, authSecret, k8sClient)
 	if err != nil {
 		return nil, fmt.Errorf("failed to load auth credentials: %w", err)
 	}
 
 	return auth, nil
 }
 
-func loadAuthCredentials(namespace, secretName string, k8sClient Client) ([]string, error) {
+func getDigestAuthCredentials(k8sClient Client, authSecret, namespace string) ([]string, error) {
+	if authSecret == "" {
+		return nil, fmt.Errorf("auth secret must be set")
+	}
+
+	auth, err := loadDigestAuthCredentials(namespace, authSecret, k8sClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load auth credentials: %w", err)
+	}
+
+	return auth, nil
+}
+
+func loadBasicAuthCredentials(namespace, secretName string, k8sClient Client) ([]string, error) {
+	secret, ok, err := k8sClient.GetSecret(namespace, secretName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch secret '%s/%s': %w", namespace, secretName, err)
+	}
+	if !ok {
+		return nil, fmt.Errorf("secret '%s/%s' not found", namespace, secretName)
+	}
+	if secret == nil {
+		return nil, fmt.Errorf("data for secret '%s/%s' must not be nil", namespace, secretName)
+	}
+	if len(secret.Data) == 2 {
+		username, usernameExists := secret.Data["username"]
+		password, passwordExists := secret.Data["password"]
+		if !(usernameExists && passwordExists) {
+			return nil, fmt.Errorf("found 2 elements for secret '%s/%s', must contain username and password keys", namespace, secretName)
+		}
+
+		if secret.Type != corev1.SecretTypeBasicAuth {
+			return nil, fmt.Errorf("secret '%s/%s', must be of type %s", namespace, secretName, corev1.SecretTypeBasicAuth)
+		}
+
+		return generateCredentialsFromUsernamePassword(username, password)
+	}
+
+	if len(secret.Data) != 1 {
+		return nil, fmt.Errorf("found %d elements for secret '%s/%s', must be single element exactly", len(secret.Data), namespace, secretName)
+	}
+
+	var firstSecret []byte
+	for _, v := range secret.Data {
+		firstSecret = v
+		break
+	}
+
+	var credentials []string
+	scanner := bufio.NewScanner(bytes.NewReader(firstSecret))
+	for scanner.Scan() {
+		if cred := scanner.Text(); len(cred) > 0 {
+			credentials = append(credentials, cred)
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("error reading secret for %s/%s: %w", namespace, secretName, err)
+	}
+	if len(credentials) == 0 {
+		return nil, fmt.Errorf("secret '%s/%s' does not contain any credentials", namespace, secretName)
+	}
+
+	return credentials, nil
+}
+
+func generateCredentialsFromUsernamePassword(username, password []byte) ([]string, error) {
+	obscuredPass, err := bcrypt.GenerateFromPassword(password, bcrypt.MinCost)
```

Comment:
Since we are loading an un-encrypted password from the api, and the auth middleware requires a hashed password, I have bcrypt generate it.

However this is not consistent, as bcrypt uses a new salt each run.

Should we replace this with a consistent hash?